File: frontend/src/api/teams.ts
import axios from 'axios';

const API_URL = 'http://localhost:8000/api/v1';

export interface TeamsSessionCreate {
    meeting_url: string;
    display_name: string;
    mic_enabled: boolean;
    camera_enabled: boolean;
    audio_input_device?: string;
    audio_output_device?: string;
    video_device?: string;
}

export interface TeamsSessionResponse {
    session_id: string;
    status: string;
    created_at: string;
}

export const teamsAPI = {
    joinSession: async (data: TeamsSessionCreate): Promise<TeamsSessionResponse> => {
        const response = await axios.post(`${API_URL}/teams/join-session`, data);
        return response.data;
    }
};

File: backend/app/schemas/teams.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class TeamsSessionCreate(BaseModel):
    meeting_url: str
    display_name: str
    mic_enabled: bool
    camera_enabled: bool
    audio_input_device: Optional[str] = None
    audio_output_device: Optional[str] = None
    video_device: Optional[str] = None

class TeamsSessionResponse(BaseModel):
    session_id: str
    status: str
    created_at: datetime

    class Config:
        from_attributes = True
